routes/videos/VideosController.cs

```diff
@@ -262,4 +262,116 @@ public async Task<IActionResult> AddVideoComment(YoutubeContext db, long videoId
 
         return Ok();
     }
+     /* [HttpGet("{videoId}/like")] //do details albo request
+ [Authorize(AuthenticationSchemes = CookieAuthenticationDefaults.AuthenticationScheme)]
+ [RequiresUserAccount]
+ public async Task<IActionResult> GetVideoLikes(YoutubeContext db, int videoID )
+ {
+     var userId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+
+     
+     var userRating = await db.Likes
+         .Where(l => l.Video == videoID && l.Account == long.Parse(userId))
+         .Select(l => new { l.Unlike })
+         .FirstOrDefaultAsync();
+
+     if (userRating == null)
+     {
+         return Ok(new { rated = false });
+     }
+
+     return Ok(new
+     {
+         rated = true,
+         dislike = userRating.Unlike // 'true' jeśli to dislike, 'false' jeśli to like
+     });
+ }
+ */
+
+     [HttpPost("{videoId}/like")]
+    [Authorize(AuthenticationSchemes = CookieAuthenticationDefaults.AuthenticationScheme)]
+    [RequiresUserAccount]
+    public async Task<IActionResult> LikeVideo(YoutubeContext db, [FromBody] AddVideoCommentModel body)
+    {
+        Video? video = await db.Videos.FindAsync((int)body.videoId);
+        if (video == null)
+        {
+            return NotFound("404");
+        }
+
+        var userId = User.FindFirst(ClaimTypes.NameIdentifier);
+
+        Like? likePrevID = await db.Likes
+        .OrderByDescending(l => l.Id)
+        .FirstOrDefaultAsync();
+
+        int likeID;
+
+        if (likePrevID == null)
+        {
+            likeID = 0;
+        }
+        else likeID=likePrevID.Id + 1;
+
+        bool likeState;
+
+        if (body.value == -1) likeState = true;
+        else likeState = false;
+
+        if (!db.Likes.Any(l => l.Video == body.videoId && l.Account == long.Parse(userId.Value!)))
+        {
+            Like like = new Like
+            {
+                Id = likeID,
+                Video = (int)body.videoId,
+                Account = long.Parse(userId.Value!),
+                Unlike = likeState
+            };
+            db.Likes.Add(like);
+            await db.SaveChangesAsync();
+
+            return Ok(new
+            {
+                id = like.Id,
+                videoId = like.Video,
+                account = like.Account,
+                unlike = like.Unlike,
+
+            });
+        }
+        else
+        {
+            var existingLike = await db.Likes.FirstOrDefaultAsync(l => l.Video == body.videoId && l.Account == long.Parse(userId.Value!));
```

Comment:
trochę też nie optymalne, ponieważ na początku sprawdzamy czy nie ma elementów a potem pobieramy element kolejnym zapytaniem, jeżeli już jakiś rekord istnieje w bazie. Nie można zrobić raz zapytania do bazy, żeby sprawdzić czy element istnieje przed ifami a potem jeżeli existingLike jest nullem to dodać nowy element, a jeżeli existingLike istnieje to go zmodyfikować i zapisać zmiany w bazie.